src/main/cheat_code.cpp

```diff
@@ -1010,739 +1088,1028 @@ void CheatCode(char NewKey)
-            cheated = true;
+            auto cc = c;
+            SDL_strlcpy(cc.key, alias.c_str(), SDL_min(sizeof(cc.key) - 1, alias.size() + 1));
+            dst->push_back(cc);
+            break;
         }
-        else if(SDL_strstr(CheatString.c_str(), "tooslow"))
+    }
+}
+
+void cheats_rename(CheatsScope scope, const std::string &source, const std::string &alias)
+{
+    std::vector<CheatCode_t> *dst = nullptr;
+
+    switch(scope)
+    {
+    case CHEAT_SCOPE_GLOBAL:
+        dst = &s_cheatsListGlobal;
+        break;
+    case CHEAT_SCOPE_WORLD:
+        dst = &s_cheatsListWorld;
+        break;
+    case CHEAT_SCOPE_LEVEL:
+        dst = &s_cheatsListLevel;
+        break;
+    }
+
+    SDL_assert(dst);
+
+    for(auto &c : *dst)
+    {
+        if(source.compare(c.key) == 0)
         {
-            if(SuperSpeed)
-            {
-                SuperSpeed = false;
-                PlaySound(SFX_PlayerShrink);
-            }
-            else
-            {
-                SuperSpeed = true;
-                PlaySound(SFX_PlayerGrow);
-            }
-            CheatString.clear();
-            cheated = true;
+            SDL_strlcpy(c.key, alias.c_str(), SDL_min(sizeof(c.key) - 1, alias.size() + 1));
+            break;
         }
-        else if(SDL_strstr(CheatString.c_str(), "ahippinandahoppin") || SDL_strstr(CheatString.c_str(), "jumpman"))
+    }
+
+}
+
+void cheats_erase(CheatsScope scope, const std::string &source)
+{
+    std::vector<CheatCode_t> *dst = nullptr;
+
+    switch(scope)
+    {
+    case CHEAT_SCOPE_GLOBAL:
+        dst = &s_cheatsListGlobal;
+        break;
+    case CHEAT_SCOPE_WORLD:
+        dst = &s_cheatsListWorld;
+        break;
+    case CHEAT_SCOPE_LEVEL:
+        dst = &s_cheatsListLevel;
+        break;
+    }
+
+    SDL_assert(dst);
+
+    for(auto it = dst->begin(); it != dst->end(); ++it)
+    {
+        if(source.compare(it->key) == 0)
         {
-            if(MultiHop)
-            {
-                MultiHop = false;
-                PlaySound(SFX_PlayerShrink);
-            }
-            else
-            {
-                MultiHop = true;
-                PlaySound(SFX_PlayerGrow);
-            }
-            CheatString.clear();
-            cheated = true;
+            dst->erase(it);
+            break;
         }
-        else if(SDL_strstr(CheatString.c_str(), "framerate"))
+    }
+}
+
+
+/*!
+ * \brief Process the cheat buffer
+ * \param NewKey New key symbol
+ */
+void CheatCode(char NewKey)
+{
+    std::string newCheat;
+    std::string oldString;
+    bool cheated = false;
+
+    if(LevelEditor || GameMenu || /*nPlay.Online ||*/ BattleMode)
+    {
+        CheatString.clear();
+        return;
+    }
+
+    CheatString.push_back(NewKey);
+    if(CheatString.size() > 23)
+        CheatString.erase(0, 1);
+    oldString = CheatString;
+
+
+    for(const auto &c : s_cheatsListGlobal)
+    {
+        if(CheatString.find(c.key) != std::string::npos)
```

Comment:
Can you do a direct suffix check here instead of the find call?

Reply:
Good idea

Reply:
And, for the best performance, reasonable to use `memcmp()` here